knowledge-base/displaying-reports-from-report-server-through-custom-report-source-resolver.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Displaying Reports From Report Server Through Custom Report Source Resolver
+description: Explains how to connect to Report Server through Custom Report Source Resolver and display a reprot
+type: how-to
+page_title: How to Display Reports From Report Server through Custom Report Source Resolver
+slug: displaying-reports-from-report-server-through-custom-report-source-resolver
+position: 
+tags: 
+ticketid: 1519100
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+Sometimes you want to avoid adding the Report Server's credentials in the initialization of the viewer. This can be done by the usage of [Custom Report Source Resolver](../telerik-reporting-rest-custom-report-resolver).
+
+## Solution
+1. You need to add references to the **Telerik.ReportServer.Services.Models** assembly.
+
+2. In the Resolve method of the CustomReportSourceResolver, firstly, you should create a setting for the Report Server connection. Then based on the passed category and report name from the viewer, you need to get the report from the Report Server and return it as an InstanceReportSource. Eventually, the CustomReportSourceResolver has to look as follows:
+
```

Comment:
I think that the last sentence is unnecessary or at least could be simplified to "For example:"